smart-contracts/contracts/cl-vault/src/vault/range.rs

```diff
@@ -189,6 +176,19 @@ pub fn handle_withdraw_position_reply(
         })
     }
 
+    Ok(AmountsAndTokensProvidedResponse {
```

Comment:
Why do we return amount0 and amount1 if that is also provided through the tokens_provided?
Not sure about all the context, to me it looks like 
* either we should just return tokens_provided
* or we should just return amount0 and amount1 and construct tokens_provided elsewhere

Reply:
i.e. `do_swap_deposit_merge` function seeks for a tuple of both of them` (balances.amount0, balances.amount1)`. If we refactor all the other functions to work with tokens_provided Vec this would indeed be possible.

Reply:
Addressed by splitting the getter helper in two different functions. So i got rid of the new struct i was introducing. Looks cleaner now, thanks!